ompi/mca/mtl/ofi/mtl_ofi.h

```diff
@@ -110,10 +110,15 @@ ompi_mtl_ofi_progress(void)
                 abort();
             }
         } else {
-            /**
-             * The CQ is empty. Return.
-             */
-            break;
+            /* CQ is empty, return */
+            if (ret == -FI_EAGAIN) {
+                break;
+            } else {
+                opal_output_verbose(1, ompi_mtl_base_framework.framework_output,
+                                    "%s:%d: fi_cq_read failed: %s(%zd)\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret), ret);
```

Comment:
This message should be output via `opal_show_help()`, not `opal_output_verbose()`.

Reply:
Other parts of the code emit error messages in the same manner- either using opal_output_verbose() or opal_output() (in ompi_mtl_ofi_progress()). Would you prefer I convert all these instances to use opal_show_help() as well? Also, general clarification for future- when exactly should we use opal_show_help() vs. opal_output()/_verbose() ?

Reply:
If you have the cycles to do so, yes -- converting all user-visible errors to use `opal_show_help()` would be great.

In general, we've all be a bit sloppy about using `opal_output()` and `opal_output_verbose()` to show user errors.  I try to catch PRs that add *new* error messages via `opal_output[_verbose]()` and ask them to update to `opal_show_help()`.

In reality, you should classify each message and decide what you want:

1. If it's a warning or an error message that needs to *always* be displayed to the user, then it should use `opal_show_help()`.  Particularly if it can happen on multiple processes in the same job (because `opal_show_help()` de-duplicates and only shows the message once).
1. If it's a behind-the-scenes message that is really for developers (or when you're troubleshooting with a user), then `opal_output_verbose()` is sufficient.  I.e., you can show the message *sometimes*, but basically only when it is specifically requested by the user (by setting a verbosity level).

Does that make sense / help?

Reply:
Yes, makes sense. Thanks for the explanation. Shall try to keep this rule of thumb in mind when adding more messages.

Reply:
Hi @jsquyres, in this particular case if reading fi_cq fails the mtl progress function will return an error that should be captured by opal_progress. So, wouldn't opal_progress() actually print the user friendly message?  My point is that there is not a clear line (I mean, I don't have it)  on what is important to the user vs developer for this case. It seems to me that the fi_errno is more like a developers thing. Clarification ? thanks!

Reply:
Yeah, we're probably not very consistent across the code base in this regard (i.e., what to do if a component `progress()` function fails) -- there's probably no hard-and-fast rule here.

My $0.02: if you know the job is going to abort, `opal_show_help()` something useful in friendly language that a user can understand.